tchannel/peer_strategy.py

```diff
@@ -30,3 +32,24 @@ def get_score(self, peer):
 class ZeroScoreCalculator(ScoreCalculator):
     def get_score(self, peer):
         return 0
+
+
+class PreferIncomingCalculator(ScoreCalculator):
+    def get_score(self, peer):
+        """Calculate the peer score based on connections.
+
+        If the peer has no incoming connections, it will have largest score.
+        In our peer selection strategy, the largest number has least priority
+        in the heap.
+
+        If the peer has incoming connections, we will return number of outbound
+        pending requests and responses.
+
+        :param peer: instance of `tchannel.tornado.peer.Peer`
+        :return: score of the peer
+        """
+        if (peer.host == '0.0.0.0' or int(peer.port) == 0 or
+                not peer.incoming_connections):
```

Comment:
Skipping peers that don't have any incoming connections seems risky. I'm imagining a scenario where you have one incoming connection and N outgoing connections. This will cause all the load to be sent to that one incoming connection. We shouldn't tie the scoring behavior to how Hyperbahn works right now.